src/commands/mapsheet-coverage/mapsheet.coverage.ts

```diff
@@ -0,0 +1,212 @@
+      captureArea.properties = captureArea.properties ?? {};
+      captureArea.properties['source'] = targetCaptureAreaUrl.href;
+      captureArea.properties['title'] = collection.title;
+      for (const [key, value] of Object.entries(collection)) {
+        if (key.startsWith('linz:')) captureArea.properties[key] = value;
+      }
+      logger.debug(
+        {
+          layer: layer.name,
+          title: collection.title,
+          layerSizeBytes: Math.round(JSON.stringify(captureArea.geometry).length),
+        },
+        'Processing',
+      );
+
+      // GeoJSON over about 8 decimal places starts running into floating point math errors
+      truncateGeoJson(captureArea);
+
+      // Determine if this layer has any additional information to the existing layers
+      const diff = pc.difference(captureArea.geometry.coordinates as pc.MultiPolygon, total);
+
+      if (diff.length === 0) {
+        // Layer has no information included in the output
+        logger.warn({ layer: layer.name, location: targetCaptureAreaUrl.href }, 'FullyCovered');
+        await fsa.write(fsa.join(OutputPath, `remove-${layer.name}.geojson`), JSON.stringify(captureArea));
+        // break;
+        continue;
+      }
+
+      // Extract all the links from the collection and assign them to each mapsheet
+      // This assumes that tiles are in the format "./BX01_1000_0101.json"
+      for (const link of collection.links) {
+        if (link.rel !== 'item') continue;
+        const url = new URL(link.href, targetCollection);
+        const fileName = basename(url.pathname);
+
+        const ms = MapSheet.getMapTileIndex(fileName);
+        if (ms == null) throw new Error('Unable to extract mapsheet from ' + url.href);
+        const existing = mapSheets.get(ms.mapSheet) ?? [];
+        existing.push(url.href);
+        mapSheets.set(ms.mapSheet, existing);
+      }
+
+      total = pc.union(total, captureArea.geometry.coordinates as pc.MultiPolygon);
+      previous.features.push({
+        type: 'Feature',
+        geometry: { type: 'MultiPolygon', coordinates: diff },
+        properties: captureArea.properties,
+      });
+      geojson.features.push(captureArea);
+    }
+
+    // All the source layers as a single file
+    logger.info('Write:SourceFeatures');
+    await fsa.write(fsa.join(OutputPath, 'layers-source.geojson.gz'), gzipSync(JSON.stringify(geojson)));
+
+    // A single output feature for total capture area
+    logger.info('Write:CombinedUnion');
+    await fsa.write(
+      fsa.join(OutputPath, 'layers-combined.geojson.gz'),
+      gzipSync(JSON.stringify({ type: 'Feature', geometry: { type: 'MultiPolygon', coordinates: total } })),
+    );
+
+    // Which areas of each layers are needed for the output
+    logger.info('Write:RequiredLayers');
+    await fsa.write(fsa.join(OutputPath, 'layers-required.geojson.gz'), gzipSync(JSON.stringify(previous)));
+
+    // List of files to be created
+    const todo = [...mapSheets].map((m) => {
```

Comment:
this is in the wrong priority order, the files list be reversed for gdal to laeyr them correctly